### PR TITLE
fix(cmake): added zig workarounds for libelf and grpc cmake modules.

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

/area build

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

Moreover, patch grpc cmake module to patch its abseil-cpp submodule to be compatible with gcc-13.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

Last patch for https://github.com/falcosecurity/falco/pull/3307.
Actually, extracted from it.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/milestone 0.19.0

### CODE REVIEW

**Comment by FedeDP** [comment]
In a follow-up PR i will also add a zig CI for libs leveraging the Falco composite action to install zig: https://github.com/falcosecurity/falco/pull/3307/files#diff-4272283b257713d488f43b293b0303ea8d82c0aaa72265de68b5eebccde26b29

File: cmake/modules/libelf.cmake [hunk 1]
```diff
@@ -25,15 +25,31 @@ elseif(NOT USE_BUNDLED_LIBELF)
...
+			configure_file(${LIBELF_INCLUDE}/gelf.h libelf/gelf.h COPYONLY)
+			set(LIBELF_INCLUDE ${CMAKE_CURRENT_BINARY_DIR}/libelf)
+		endif()
 		message(STATUS "Found LIBELF: include: ${LIBELF_INCLUDE}, lib: ${LIBELF_LIB}")
 	else()
 		message(FATAL_ERROR "Couldn't find system libelf")
 	endif()
-	# We add a custom target, in this way we can always depend on `libelf` without distinguishing
-	# between "bundled" and "not-bundled" case
-	add_custom_target(libelf)
```
Comment (FedeDP) [inline cmake/modules/libelf.cmake:36]: No need for this since it is already present at the end of the file.

**Review by LucaGuerra — APPROVED** [review]

**Review by leogr — APPROVED** [review]

**Pull request merged**